### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by faridahyderi**

### CODE REVIEW

File: 02-express-tutorial/app.js [hunk 1]
```diff
@@ -1 +1,107 @@
...
+
+// Creation of the app as returned from calling express()
+const app = express();
+
+//app.use statements for the middleware. You’ll eventually use many kinds of middleware, 
+//but for now the only middleware we are using is express.static().
+app.use(express.static("./public"))
+
+//Add an app.get statement to app.js. It should be after the Express static middleware, but before the “not found” handler. 
+/*app.get('/api/v1/test',(req,res)=>{
```
Comment (Coding-Algorithm) [inline 02-express-tutorial/app.js]: uncomment

File: 02-express-tutorial/app.js [hunk 2]
```diff
@@ -1 +1,107 @@
...
+
+app.get('/api/v1/products',(req,res)=>{
+    res.json(products);
+})
+
+/*So add an app.get statement for the url
+/api/v1/products. Write some code to return JSON for the products
+array.*/
+/*app.get('/api/v1/products/:productID',(req,res)=>{
+
```
Comment (Coding-Algorithm) [inline 02-express-tutorial/app.js:28]: Welldone. 
But I noticed you did not handle the stituation where the productID is not found.

Kindly do that and uncomment the code

File: 02-express-tutorial/app.js [hunk 3]
```diff
@@ -1 +1,107 @@
...
+    );
+
+    let limitedSearch = simpleSearch.slice(0,limit);
+    res.json(limitedSearch);
+
+})*/
+
+// the user might want to send a regular expression instead of search for starting letters
+/*app.get('/api/v1/query', (req,res)=>{
+    console.log("inside new get")
```
Comment (Coding-Algorithm) [inline 02-express-tutorial/app.js:57]: Good job.. 

Uncomment your code..

**Review by Coding-Algorithm — CHANGES_REQUESTED** [review]
Check individual comments for adjustments

File: 02-express-tutorial/app.js (hunk 3 above)
Comment (faridahyderi) [reply 02-express-tutorial/app.js:57]: Please check, i believe i resolved.

> Good job..
> 
> Uncomment your code..

``

File: 02-express-tutorial/app.js (hunk 2 above)
Comment (faridahyderi) [reply 02-express-tutorial/app.js:28]: Please check, i believe i resolved.

> Welldone. But I noticed you did not handle the stituation where the productID is not found.
> 
> Kindly do that and uncomment the code

**Pull request merged**